### ISSUE TITLE
Run train op multiple times

### ISSUE THREAD

**Issue by vladfi1** [tensorflow/tensorflow#2764]
I have some fairly large batch sizes on which I'd like to take multiple gradient steps. While I could easily do this with a python for loop, I imagine that there might be a more efficient method that doesn't involve transferring the data to gpu on each iteration.

@yaroslavvb has confirmed that putting the train op in the fetch list multiple times doesn't work, and suggested that I could save my input tensors as variables with `tf.assign`. Unfortunately, this doesn't work for me because my input tensors have variable sizes, and tensorflow variables must have a fixed size.

**Comment by yaroslavvb** [comment]
Updated [SO discussion](http://stackoverflow.com/questions/37644984/run-train-op-multiple-times-in-tensorflow/37645647?noredirect=1#comment62941927_37645647) with another suggestion

**Comment by vladfi1** [comment]
Your second answer solved my problem. Thanks for the help!

**Comment by yaroslavvb** [comment]
For future reference, the solution was to use @yuanbyu 's persistent tensors to persist the input tensors between run calls
